<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from seastar.urdf.xacro             | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="seastar">
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="orange">
    <color rgba="1 0.3 0.1 1"/>
  </material>
  <material name="blue">
    <color rgba="0.2 0.2 1 1"/>
  </material>
  

!-- the IMU is fixed to the base of the robot --&gt;

  <link name="imu_link">
</link>
  <joint name="imu_to_base" type="fixed">
    <parent link="imu_link"/>
    <child link="base_link"/>
    <origin rpy="0 0 3.141592653589793" xyz="0 0 0"/>
  </joint>
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0062"/>
      <geometry>
        <box size="0.6223 0.1 0.0124"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.31115 0.05 0.0062"/>
      <geometry>
        <box size="0.6223 0.1 0.0124"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.31115 0.05 0.0062"/>
      <mass value="0.5"/>
      <inertia ixx="0.00042307333333333337" ixy="0.0" ixz="0.0" iyy="0.016142127083333332" iyz="0.0" izz="0.01655238708333333"/>
    </inertial>
  </link>
  <!-- base to upright -->
  <joint name="base_to_upright1_joint" type="fixed">
    <parent link="base_link"/>
    <child link="upright1_link"/>
    <origin rpy="0 -1.5707963267948966 0" xyz="0.31115 0 0"/>
  </joint>
  <link name="upright1_link">
    <visual>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <geometry>
        <box size="0.3 0.1 0.0125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <geometry>
        <box size="0.3 0.1 0.0125"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.0004231770833333334" ixy="0.0" ixz="0.0" iyy="0.0037565104166666662" iyz="0.0" izz="0.004166666666666667"/>
    </inertial>
  </link>
  <!-- upright to roll -->
  <joint name="upright1_to_roll_joint" type="revolute">
    <parent link="upright1_link"/>
    <child link="roll_link"/>
    <origin rpy="0 -1.5707963267948966 0" xyz="0.3 0 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="100" lower="-3.64774" upper="0.558505" velocity="100"/>
  </joint>
  <link name="roll_link">
    <visual>
      <origin rpy="0 0 0" xyz="0.3 0 0"/>
      <geometry>
        <box size="0.6 0.1 0.0125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.3 0 0"/>
      <geometry>
        <box size="0.6 0.1 0.0125"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.3 0 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.0004231770833333334" ixy="0.0" ixz="0.0" iyy="0.015006510416666667" iyz="0.0" izz="0.015416666666666665"/>
    </inertial>
  </link>
  <!-- roll to pitch-->
  <joint name="roll_to_pitch_joint" type="revolute">
    <parent link="roll_link"/>
    <child link="pitch_link"/>
    <origin rpy="0 -1.5707963267948966 0" xyz="0.3 0 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="100" lower="3.141592653589793" upper="7.34784" velocity="100"/>
  </joint>
  <link name="pitch_link">
    <visual>
      <origin rpy="0 0 0" xyz="0.1 0 0"/>
      <geometry>
        <box size="0.2 0.1 0.0125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.1 0 0"/>
      <geometry>
        <box size="0.2 0.1 0.0125"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.1 0 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.0004231770833333334" ixy="0.0" ixz="0.0" iyy="0.0016731770833333334" iyz="0.0" izz="0.0020833333333333337"/>
    </inertial>
  </link>
  <!-- pitch to stokes -->
  <joint name="pitch_to_stokes_joint" type="revolute">
    <parent link="pitch_link"/>
    <child link="stokes_link"/>
    <origin rpy="0 -1.5707963267948966 0" xyz="0.1 0 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="100" lower="-3.141592653589793" upper="3.141592653589793" velocity="100"/>
  </joint>
  <link name="stokes_link">
    <visual>
      <origin rpy="0 1.5707963267948966 0" xyz="0.1 0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.06"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.1 0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.06"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.1 0 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.002116666666666667" ixy="0.0" ixz="0.0" iyy="0.002116666666666667" iyz="0.0" izz="0.0009"/>
    </inertial>
  </link>
</robot>
